### PR TITLE
feat(app): add protocol overview section

### LINKED ISSUE #8821: Protocol Overview Section
# Overview
The top section of the protocol details page is an overview of the protocol (including protocol name, creation method, estimated run time, last updated, last analyzed (Python only), org/author, and description. 

# Design
- This artboard contains: The design of this overview section, including the expanded description field and an example of how additional Python metadata fields will render  https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1482%3A105272

# Implementation Details
- Python protocols can contain custom key-value metadata pairs that we should render in the overview section. PD protocols will not have these fields.

# Acceptance Criteria
- [ ] Protocol metadata from PD and Python protocols render correctly in the UI

- [ ] Clicking the "Run Protocol" button opens up the Robot Selection slide-out

# Selectors for E2E

- [ ] protocol name
- [ ] creation method
- [ ] estimated run time
- [ ] last updated
- [ ] last analyzed (Python only)
- [ ] org/author
- [ ] description
- [ ] Overflow button
- [ ] Read less link

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR adds the protocol overview section in the protocol details page. closes #8821

Screen Shot 2022-05-04 at 5 09 03 PM

Screen Shot 2022-05-04 at 6 13 05 PM

# Changelog

- Added protocol details to the `ProtocolDetails` component.
- Updated overflow menu
- Added `Read More` and `Read Less` logic to the description.

# Review requests

- Upload a JSON and/or Python protocol and check that the protocol details are rendering correctly.

# Risk assessment
low

### CODE REVIEW

**Review by jerader — APPROVED** [review]
looks great! Tested a few protocols as well. ⭐

File: app/src/organisms/ProtocolDetails/index.tsx [hunk 1]
```diff
@@ -104,6 +106,83 @@ function RoundTab({
...
+    if (protocolType === 'json') {
+      return description
+    } else {
+      const filteredMetaData = Object.entries(metadata)
+        .map(e => ({
+          [e[0]]: e[1],
+        }))
+        .filter(
+          item =>
+            Object.keys(item)[0] !== 'protocolName' &&
```
Comment (sakibh) [inline app/src/organisms/ProtocolDetails/index.tsx]: use an array and `includes` method to check for these keys.

Comment (brenthagen) [reply app/src/organisms/ProtocolDetails/index.tsx]: lodash `omit` would work well for this also

File: app/src/organisms/ProtocolDetails/index.tsx [hunk 2]
```diff
@@ -269,17 +368,15 @@ export function ProtocolDetails(
           >
             <StyledText as="h6">{t('description')}</StyledText>
             <StyledText as="p">
-              {analysisStatus === 'loading' ? t('shared:loading') : description}
+              {analysisStatus === 'loading' ? (
+                t('shared:loading')
+              ) : (
+                <ReadMoreContent
```
Comment (brenthagen) [inline app/src/organisms/ProtocolDetails/index.tsx]: nesting this component within a p tag (the `StyledText`  in line 370) results in HTML that isn't semantic - instead give the other half of the ternary its own `StyledText`. we have a few errors like this that show up in the console throughout the app, something to watch out for

File: app/src/organisms/ProtocolDetails/index.tsx [hunk 3]
```diff
@@ -147,11 +226,26 @@ export function ProtocolDetails(
...
-  const description = t('shared:no_data')
-  const lastAnalyzed = t('shared:no_data')
+  const getCreationMethod = (config: JsonConfig | PythonConfig): string => {
+    if (config.protocolType === 'json') {
+      return t('protocol_designer_version', {
+        version: config.schemaVersion.toFixed(1),
+      })
+    } else {
+      return t('python_api_version', {
+        version: config.apiVersion.toString().replace(',', '.'),
```
Comment (brenthagen) [inline app/src/organisms/ProtocolDetails/index.tsx]: i get a white screen from the toString call here on a protocol with a failed analysis, we should use optional chaining at least

File: app/src/organisms/ProtocolDetails/index.tsx [hunk 4]
```diff
@@ -147,11 +226,26 @@ export function ProtocolDetails(
...
+    }
+  }
+
+  const creationMethod =
+    getCreationMethod(mostRecentAnalysis.config) ?? t('shared:no_data')
+  const author = isEmpty(mostRecentAnalysis.metadata.author)
+    ? t('shared:no_data')
+    : mostRecentAnalysis.metadata.author
+  const lastAnalyzed =
+    format(new Date(mostRecentAnalysis.createdAt), 'MMMM dd, yyyy HH:mm') ??
```
Comment (brenthagen) [inline app/src/organisms/ProtocolDetails/index.tsx]: i also get a whitescreen here from the format call with a failed analysis: 
```    
mostRecentAnalysis.createdAt != null
      ? format(new Date(mostRecentAnalysis.createdAt), 'MMMM dd, yyyy HH:mm')
      : t('shared:no_data')
```

**Review by brenthagen — COMMENTED** [review]
looks good! a couple of comments on whitescreens from protocols with failed analyses - we should consider protecting the other date-fns `format` calls as well.

**Review by brenthagen — APPROVED** [review]
guessing the test failure is a timezone issue - probably best to remove that line

**Pull request merged**